src/cls/tabular/fbreader.cc

```diff
@@ -0,0 +1,256 @@
+      for( unsigned int i = 0; i < rows_data->Length(); i++ ) {
+        if( debug ) {
+          std::cout << rids_read->Get(i) << ":\t" ;
+        }
+
+        auto curr_rec           = rows_data->Get(i) ;
+        auto curr_rec_data      = curr_rec->data() ;
+        auto curr_rec_data_type = curr_rec->data_type() ;
+
+        //std::cout << "curr_rec_data->Length() = " << curr_rec_data->Length() << std::endl ;
+
+        for( unsigned int j = 0; j < curr_rec_data->Length(); j++ ) {
+          // column of ints
+          if( (unsigned)curr_rec_data_type->Get(j) == Tables::Data_IntData ) {
+            //std::cout << "int" << "\t" ;
+            auto int_col_data = static_cast< const Tables::IntData* >( curr_rec_data->Get(j) ) ;
+            std::cout << int_col_data->data()->Get(0) ;
+          }
+          // column of floats
+          else if( (unsigned)curr_rec_data_type->Get(j) == Tables::Data_FloatData ) {
+            //std::cout << "float" << "\t" ;
+            auto float_col_data = static_cast< const Tables::FloatData* >( curr_rec_data->Get(j) ) ;
+            std::cout << float_col_data->data()->Get(0) ;
+          }
+          // column of strings
+          else if( (unsigned)curr_rec_data_type->Get(j) == Tables::Data_StringData ) {
+            //std::cout << "str" << "\t" ;
+            auto string_col_data = static_cast< const Tables::StringData* >( curr_rec_data->Get(j) ) ;
+            std::cout << string_col_data->data()->Get(0)->str() ;
+          }
+          else {
+            std::cout << "execute_query: unrecognized row_data_type " 
+                      << (unsigned)curr_rec_data_type->Get(j) << std::endl ; 
+          }
+
+          if( j < curr_rec_data->Length()-1 )
+            std::cout << "," ;
+        }
+        std::cout << std::endl ;
+      }
+    } // Relation_Rows 
+
+    // process one Root>Col flatbuffer
+    else if( data_type == Tables::Relation_Col ) {
+      if( debug ) {
+        std::cout << "else if data_type == Tables::Relation_Col" << std::endl ;
+      }
+
+      auto col = static_cast< const Tables::Col* >( root->relationData() ) ;
+      auto col_name_read  = col->col_name() ;
+      auto col_index_read = col->col_index() ;
+      auto nrows_read     = col->nrows() ;
+      auto rids_read      = col->RIDs() ;
+      auto col_data_type  = col->data_type() ;
+      auto col_data       = col->data() ;
+
+      if( debug ) {
+        std::cout << "col_name_read->str() : " << col_name_read->str() << std::endl ;
+        std::cout << "col_index_read       : " << (unsigned)col_index_read << std::endl ;
+        std::cout << "nrows_read           : " << (unsigned)nrows_read     << std::endl ;
+        std::cout << "col_data_type        : " << col_data_type     << std::endl ;
+      }
+
+      // print data to stdout
+      for( unsigned int i = 0; i < nrows_read; i++ ) {
+        if( debug ) {
+          std::cout << rids_read->Get(i) << ":\t" ;
+        }
+        // column of ints
```

Comment:
for clarity can you please change these if/else ifs to switch((unsigned)col_data_type) {case Tables::Data_IntData, etc.